core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java

```diff
@@ -0,0 +1,345 @@
+                Long256 value = args.getQuick(i).getLong256A(rec);
+                if (isNotNull(value)) {
+                    Numbers.appendLong256(value.getLong0(), value.getLong1(), value.getLong2(), value.getLong3(), sink);
+                    return;
+                }
+            }
+        }
+
+        @Override
+        public Long256 getLong256A(Record rec) {
+            Long256 value = Long256Impl.NULL_LONG256;
+            for (int i = 0, size = args.size(); i < size; i++) {
+                value = args.getQuick(i).getLong256A(rec);
+                if (isNotNull(value)) {
+                    return value;
+                }
+            }
+            return value;
+        }
+
+        @Override
+        public Long256 getLong256B(Record rec) {
+            Long256 value = Long256Impl.NULL_LONG256;
+            for (int i = 0, size = args.size(); i < size; i++) {
+                value = args.getQuick(i).getLong256B(rec);
+                if (isNotNull(value)) {
+                    return value;
+                }
+            }
+            return value;
+        }
+
+        private static boolean isNotNull(Long256 value) {
+            return value != null &&
+                    value != Long256Impl.NULL_LONG256 && (value.getLong0() != Numbers.LONG_NaN ||
+                    value.getLong1() != Numbers.LONG_NaN ||
+                    value.getLong2() != Numbers.LONG_NaN ||
+                    value.getLong3() != Numbers.LONG_NaN);
+        }
+    }
+
+    private static class IntCoalesceFunction extends IntFunction implements MultiArgFunction {
+        private final ObjList<Function> args;
+
+        public IntCoalesceFunction(int position, ObjList<Function> args) {
+            super(position);
+            this.args = args;
+        }
+
+        @Override
+        public ObjList<Function> getArgs() {
+            return args;
+        }
+
+        @Override
+        public int getInt(Record rec) {
+            for (int i = 0, size = args.size(); i < size; i++) {
+                int value = args.getQuick(i).getInt(rec);
+                if (value != Numbers.INT_NaN) {
+                    return value;
+                }
+            }
+            return Numbers.INT_NaN;
+        }
+    }
+
+    private static class SymStrCoalesceFunction extends StrFunction implements MultiArgFunction {
+        private final ObjList<Function> args;
+
+        public SymStrCoalesceFunction(int position, ObjList<Function> args) {
+            super(position);
+            this.args = args;
+        }
+
+        @Override
+        public ObjList<Function> getArgs() {
+            return args;
+        }
+
+
+        @Override
+        public CharSequence getStr(Record rec) {
+            for (int i = 0, size = args.size(); i < size; i++) {
+                Function arg = args.getQuick(i);
+                CharSequence value;
+                if (arg.getType() == ColumnType.SYMBOL) {
```

Comment:
a suggestion: perhaps we can detect if all types are homogeneous or not and have more implementation for:
- all strings
- all symbols
- mixed

I think it is less likely different types are mixed. I'm open to discussion.

Reply:
done